tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/passes/ConfigurationPassImpl.scala

```diff
@@ -86,35 +90,66 @@ class ConfigurationPassImpl @Inject()(val options: WriteablePassOptions,
+          logger.info(s"  > Using the init predicate $init from the TLC config")
+          options.set("checker.init", init)
+
+        case Some(cmdInit) =>
+          logger.warn(s"  > $basename: Init operator is set both in TLC config and via --init option; using $cmdInit")
+      }
+    }
+
+    def setNext(next: String): Unit = {
+      options.get[String]("checker", "next") match {
+        case None =>
+          // In general, passes should not override options. This is a reasonable exception to this rule.
+          logger.info(s"  > Using the next predicate $next from the TLC config")
+          options.set("checker.next", next)
+
+        case Some(cmdNext) =>
+          val msg = s"  > $basename: Next operator is set both in TLC config and via --next command line option; using $cmdNext"
+          logger.warn(msg)
+      }
+    }
+
     try {
       val config = TlcConfigParser.apply(new FileReader(filename))
       configuredModule = new TlcConfigImporter(config, new IdleTracker())(module)
       config.behaviorSpec match {
         case InitNextSpec(init, next) =>
-          if(options.getOrElse("checker","init","").isEmpty) {
-            options.set("checker.init", init)
-          }
-          else {
-            logger.warn("  > Init operator is set both in " + basename + " and via --init option; using the latter")
-          }
-          if(options.getOrElse("checker","next","").isEmpty) {
-            // In general, passes should not override options. This is a reasonable exception to this rule.
-            options.set("checker.next", next)
-          }
-          else {
-            logger.warn("  > Next operator is set both in " + basename + " and via --next option; using the latter")
-          }
-        case _ => logger.warn("  > Temporal spec found in " + basename + ", which is not yet supported. Skipping.")
+          setInit(init)
+          setNext(next)
+
+        case TemporalSpec(specName) =>
+          val (init, next) = extractFromSpec(module, basename, specName)
+          logger.info(s"  > $basename: Using SPECIFICATION $specName")
+          setInit(init)
+          setNext(next)
       }
-      if(config.invariants.nonEmpty) {
-        if(options.getOrElse("checker","inv",List()).isEmpty) {
-          // In general, passes should not override options. This is a reasonable exception to this rule.
-          options.set("checker.inv", config.invariants)
-        }
-        else {
-          logger.warn("  > Invariants are set both in " + basename + " and via --inv option; using the latter")
+      if (config.invariants.nonEmpty) {
+        options.get[List[String]]("checker", "inv") match {
+          case None =>
+            // In general, passes should not override options. This is a reasonable exception to this rule.
+            logger.info(s"  > $basename: found INVARIANTS: " + String.join(", ", config.invariants :_*))
```

Comment:
This line appears in both branches of the `match` (here and on line 141), so maybe better to just construct the message once up top then log it at the appropriate level in each branch. This Will help make clear the few parts that are really essentially different in the two branches.

Reply:
Yep, fixed that